src/coverage.ts

```diff
@@ -50,21 +88,44 @@ export function initializeCodeCoverageFeatures(
 // global decoration arrays
 let coveredDecorations: vscode.DecorationOptions[] = [];
 let uncoveredDecorations: vscode.DecorationOptions[] = [];
+let partiallyCoveredDecorations: vscode.DecorationOptions[] = [];
+let coveredDecorationsWithMCDC: vscode.DecorationOptions[] = [];
+let uncoveredDecorationsWithMCDC: vscode.DecorationOptions[] = [];
+let partiallyCoveredDecorationsWithMCDC: vscode.DecorationOptions[] = [];
 
 function addDecorations(
   activeEditor: vscode.TextEditor,
   covered: number[],
-  uncovered: number[]
+  uncovered: number[],
+  partiallyCovered: number[]
 ) {
   const lineCount = activeEditor.document.lineCount;
   let lineIndex;
   // these are lists of line numbers
 
   for (lineIndex = 0; lineIndex < lineCount; lineIndex++) {
```

Comment:
The conditions are a bit convoluted and you have some repetition and a bunch of comments. You can try something like the following, but do double-check that the logic still holds and that your tests are ok

```javascript
for (lineIndex = 0; lineIndex < lineCount; lineIndex++) {
    const lineNumber = lineIndex + 1;
    const isMCDCLine = currentActiveUnitMCDCLines.includes(lineNumber);

    if (partiallyCovered.includes(lineNumber)) {
        (isMCDCLine ? partiallyCoveredDecorationsWithMCDC : partiallyCoveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }

    if (covered.includes(lineNumber)) {
        (isMCDCLine ? coveredDecorationsWithMCDC : coveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }

    if (uncovered.includes(lineNumber)) {
        (isMCDCLine ? uncoveredDecorationsWithMCDC : uncoveredDecorations)
            .push(getRangeOption(lineIndex));
        continue;
    }
}

```

Reply:
Looks better and works. Refactored in 74cbc4ab5cb5196f0b6cec7aff88c11823f60b12.